code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -563,23 +581,28 @@
 					var/datum/robot_component/C = components[V]
 					if(C.installed == 1 || C.installed == -1)
 						removable_components += V
-
+				removable_components |= stock_parts
 				var/remove = input(user, "Which component do you want to pry out?", "Remove Component") as null|anything in removable_components
-				if(!remove)
+				if(!remove || !opened || !(remove in (stock_parts|components)))
 					return
-				var/datum/robot_component/C = components[remove]
-				var/obj/item/robot_parts/robot_component/I = C.wrapped
-				to_chat(user, "You remove \the [I].")
-				if(istype(I))
-					I.brute = C.brute_damage
-					I.burn = C.electronics_damage
-
-				I.forceMove(loc)
-
-				if(C.installed == 1)
-					C.uninstall()
-				C.installed = 0
-
+				var/obj/item/removed_item
+				if(istype(components[remove], /datum/robot_component))
+					var/datum/robot_component/C = components[remove]
+					var/obj/item/robot_parts/robot_component/I = C.wrapped
+					if(istype(I))
+						I.brute = C.brute_damage
+						I.burn = C.electronics_damage
+
+					removed_item = I
+					if(C.installed == 1)
+						C.uninstall()
+					C.installed = 0
+				else if(istype(remove, /obj/item/stock_parts))
+					stock_parts -= remove
+					removed_item = remove
```

Comment:
What if it's the matter bin? You haven't cleared the separate ref to it in that case.

Reply:
Axed special handling for bin, proc that was using that var now just checks stock parts list for bins and uses their ratings.